delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/multifetch/MultiFetch.scala

```diff
@@ -0,0 +1,53 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.multifetch
+
+import ch.epfl.bluebrain.nexus.delta.sdk.acls.AclCheck
+import ch.epfl.bluebrain.nexus.delta.sdk.identities.model.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdContent
+import ch.epfl.bluebrain.nexus.delta.sdk.multifetch.model.MultiFetchResponse.Result
+import ch.epfl.bluebrain.nexus.delta.sdk.multifetch.model.MultiFetchResponse.Result._
+import ch.epfl.bluebrain.nexus.delta.sdk.multifetch.model.{MultiFetchRequest, MultiFetchResponse}
+import ch.epfl.bluebrain.nexus.delta.sdk.permissions.Permissions.resources
+import monix.bio.UIO
+
+/**
+  * Allows to fetch multiple resources of different types in one request.
+  *
+  * The response includes a resources array that contains the resources in the order specified in the request. If there
+  * is a failure getting a particular resource, the error is included in place of the resource.
+  */
+trait MultiFetch {
+
+  def apply(request: MultiFetchRequest)(implicit caller: Caller): UIO[MultiFetchResponse]
+
+}
+
+object MultiFetch {
+  def apply(
+      aclCheck: AclCheck,
+      fetchResource: MultiFetchRequest.Input => UIO[Option[JsonLdContent[_, _]]]
+  ): MultiFetch =
+    new MultiFetch {
+      override def apply(request: MultiFetchRequest)(implicit
+          caller: Caller
+      ): UIO[MultiFetchResponse] = {
+        val fetchAllCached = aclCheck.fetchAll.memoizeOnSuccess
```

Comment:
If you group by project then you don't have to cache everything, which is nice

Reply:
It avoids to go through the list several times.
Also the acl check is not only done at the project level as it tries to check at the org level and the root level so this should make it a little faster as it only queries the db once

Reply:
If a cache makes sense because it does the org level check then that's fine, but avoiding going through lists multiple times isn't something we should be concerned with. It would have no performance impact and we should optimise for readability instead

Reply:
It really depends what you do when you are doing with your lists.
And doing multiple passes can hurt readibility too, there is not a golden rule for this.
Here doing it in a single pass with the `traverse` also guarantees that the items are kept in order